### PR TITLE
Add test for Tool::getRemoteAddr

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add test for Tool::getRemoteAddr. This should help us validate https://github.com/PrestaShop/PrestaShop/issues/21780
| Type?         | improvement
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21784)

### CODE REVIEW

**Review by Progi1984 — APPROVED** [review]

File: tests-legacy/Unit/Classes/ToolsCoreTest.php [hunk 1]
```diff
@@ -454,6 +454,44 @@ public function testCeilf(float $expectedResult, float $value, int $precision)
...
+                [
+                    ["HTTP_X_FORWARDED_FOR" => "1.2.3.4,5.6.7.8", "REMOTE_ADDR" => "5.6.7.8"],
+                    '5.6.7.8'
+                ],
+                [
+                    ["REMOTE_ADDR" => "1.2.3.4"],
+                    '1.2.3.4'
+                ],
+                [
+                    ["HTTP_X_FORWARDED_FOR" => "1.2.3.4,10.2.3.4", "REMOTE_ADDR" => "10.2.3.4"],
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Classes/ToolsCoreTest.php:489]:
```suggestion
                    ["HTTP_X_FORWARDED_FOR" => "1.2.3.4,5.6.7.8", "REMOTE_ADDR" => "5.6.7.8"],
                    '5.6.7.8'
                ],
                [
                    ["REMOTE_ADDR" => "1.2.3.4"],
                    '1.2.3.4'
                ],
                [
                    ['HTTP_X_FORWARDED_FOR' => '1.2.3.4,10.2.3.4', 'REMOTE_ADDR' => '10.2.3.4'],
```

File: tests-legacy/Unit/Classes/ToolsCoreTest.php [hunk 2]
```diff
@@ -454,6 +454,44 @@ public function testCeilf(float $expectedResult, float $value, int $precision)
...
+    public function getServerHeaders()
+    {
+        return
+            [
+                [
+                    ["HTTP_X_FORWARDED_FOR" => "1.2.3.4,5.6.7.8", "REMOTE_ADDR" => "5.6.7.8"],
+                    '5.6.7.8'
+                ],
+                [
+                    ["REMOTE_ADDR" => "1.2.3.4"],
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Classes/ToolsCoreTest.php:485]:
```suggestion
                    ['REMOTE_ADDR' => '1.2.3.4'],
```

File: tests-legacy/Unit/Classes/ToolsCoreTest.php [hunk 3]
```diff
@@ -454,6 +454,44 @@ public function testCeilf(float $expectedResult, float $value, int $precision)
...
+            unset($_SERVER[$key]);
+        }
+    }
+
+    public function getServerHeaders()
+    {
+        return
+            [
+                [
+                    ["HTTP_X_FORWARDED_FOR" => "1.2.3.4,5.6.7.8", "REMOTE_ADDR" => "5.6.7.8"],
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Classes/ToolsCoreTest.php:481]:
```suggestion
                    ['HTTP_X_FORWARDED_FOR' => '1.2.3.4,5.6.7.8', 'REMOTE_ADDR' => '5.6.7.8'],
```

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
A lot of tests are missing.
- Empty remote_addr and x forwarded for
- remote_addr in the first preg_match,
- remote_addr in the second preg_match
- the third
- ...
- Only one value in the X forwarded for

**Pull request closed without merging**